stock_move_location/models/stock_picking.py

```diff
@@ -0,0 +1,40 @@
+# Copyright Jacques-Etienne Baudoux 2016 Camptocamp
+# Copyright Iryna Vyshnevska 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def button_fillwithstock(self):
+        # check source location has no children, i.e. we scanned a bin
+
+        self.ensure_one()
+        if self.location_id.child_ids:
+            raise UserError(_('Please choose a source end location'))
+        if self.move_lines:
+            raise UserError(_('Moves lines already exists'))
+        quants = self.env['stock.quant'].search(
+            [
```

Comment:
Just last thing. Could you put domain retrieval in a separate function allowing override ? Same for create (a _prepare function). Thanks

Reply:
thanks, added with git amend